File: LeetCode/easy/Count and Say.cpp
class Solution {
public:
    string countAndSay(int n) {
        if (n == 1) return "1";
        int count = 0;
        char b;
        string a = "1";
        n--;
        while (n--) {
            int length = a.size();
            string c = a;
            a = "";
            int i, j;
            for (i = 0; i < length; i++) {
                b = c[i];
                count = 1;
                for (j = i+1; j < length; j++) {
                    if (c[j] == c[j-1]) count++;
                    else break;
                }
                i += count-1;
                a += to_string(count) + b;
            }
        }
        return a;
    }
};

File: Lintcode/420_Count and Say/Count and Say.java
public class Solution {
    /*
     * @param n: the nth
     * @return: the nth sequence
     */
    public String countAndSay(int n) {
            String output = "";
            String input = "1";
            for (int i = 0; i < n-1; i++) {
                    int num = 0;
                    char temp;
                    for (int j = 0; j < input.length(); j++) {
                            num++;
                            temp = input.charAt(j); 
                            if (j < input.length()-1 && input.charAt(j+1) != temp || j == input.length()-1) {
                                    output += num;
                                    output += temp;
                                    num = 0;
                            }
                    }
                    input = output;
                    output = "";
            }
            return input;
    }
}